### ISSUE TITLE
[R-package]  lgb.train() ignores multiple metrics if their names are different lengths

### ISSUE THREAD

**Issue by jameslamb** [microsoft/LightGBM#2913]
## Environment info

Operating System: macOS Mojave 10.14.6

CPU/GPU model: 2.3 GHz Intel Core i5

C++/Python/R version: R 3.6.1
 
LightGBM version or commit hash: https://github.com/microsoft/LightGBM/commit/4aa888c8154301ebb6eb8730ec2dc1b58396497b

## Reproducible examples

The code below produces a model without error, but the metrics I passed are ignored.

```r
data(agaricus.train, package = "lightgbm")
data(agaricus.test, package = "lightgbm")
train <- agaricus.train
test <- agaricus.test

metrics <- c("binary_error", "auc", "binary_logloss")
  bst <- lgb.train(
    data = lgb.Dataset(
      train$data
      , label = train$label
    )
    , learning_rate = 1.0
    , nrounds = 10L
    , params = list(
      objective = "binary"
      , metric = metrics
    )
    , valids = list(
      "train" = lgb.Dataset(
        train$data
        , label = train$label
      )
    )
  )
```

The logs:

```
[LightGBM] [Warning] Unknown parameter: ,auc
[LightGBM] [Warning] Unknown parameter: ,binary_logloss
[LightGBM] [Warning] Unknown parameter: ,auc
[LightGBM] [Warning] Unknown parameter: ,binary_logloss
[LightGBM] [Warning] Starting from the 2.1.2 version, default value for the "boost_from_average" parameter in "binary" objective is true.
This may cause significantly different results comparing to the previous versions of LightGBM.
Try to set boost_from_average=false, if your old models produce bad results
[LightGBM] [Info] Number of positive: 3140, number of negative: 3373
[LightGBM] [Warning] Auto-choosing row-wise multi-threading, the overhead of testing was 0.002155 seconds.
You can set `force_row_wise=true` to remove the overhead.
And if memory is not enough, you can set `force_col_wise=true`.
[LightGBM] [Info] Total Bins 214
[LightGBM] [Info] Number of data points in the train set: 6513, number of used features: 107
[LightGBM] [Info] [binary:BoostFromScore]: pavg=0.482113 -> initscore=-0.071580
[LightGBM] [Info] Start training from score -0.071580
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
[LightGBM] [Warning] No further splits with positive gain, best gain: -inf
```

The warnings like `[Warning] Unknown parameter: ,binary_logloss` suggest to me that `lgb.params2str()` is not formatting the `metrics` argument correctly before sending parameters to the C++ side.

I think this is because that function uses `format()`. When `format()` is called on a character vector, by default it will return a character vector where all elements have been padded with whitespace so that they are as long as the longest element.

```r
format(c("auc", "binary_logloss"))
# [1] "auc           " "binary_logloss"
```

This is resulting in a parameter string like "... metric=auc           ,binary_logloss" (<- GitHub trims the whitespace, down to 1 space, but hopefully this shows the idea)

## Proposed Solution

`lgb.params2str()` should use `trimws()` to correct this padding from `format()`.  The tests on uses of m ultiple evaluation metrics should be improved to prevent this. Right now they don't check that all errors are reported in the Booster's `record_evals`, and nothing breaks because this problem shows up as only a warning.
